lib/resources/firewalld.rb

```diff
@@ -85,8 +85,8 @@ def has_port_enabled_in_zone?(query_port, query_zone = default_zone)
     end
 
     def has_rule_enabled?(rule, query_zone = default_zone)
-      rule = 'rule ' + rule
-      firewalld_command("--zone=#{query_zone} --query-rich-rule=#{rule}") == 'yes'
+      rule = "rule #{rule}" until rule.include?('rule')
```

Comment:
Should this be `unless` instead of `until`? `until` is usually reserved for loop control.

Also, in case the word "rule" could appear elsewhere in the string, we might want to be defensive about this and make this: `unless rule.start_with?('rule')`

Reply:
@adamleff `until` is also favored for negative conditions.

https://github.com/bbatsov/ruby-style-guide/blob/master/README.md#until-for-negatives

I agree with the comment about being defensive about the location of "rule" and I will plan on changing the branch when I get home from work.

Reply:
True, but again, it's in the context of loop control (`while` and `until`). `unless` is the preferred statement in this case, and would be consistent with the rest of the codebase.

Think of it this way: `until` is the opposite of `while`, like `unless` is the opposite of `if`.